### PR TITLE
Remove python installation as part of Linux uninstall process, Added Gnome and wget for Linux installation and .bashrc file change.

### PR DESCRIPTION
**Pull request by anssakthi**
Pull request for the implementation of removal of python installation as part of Linux uninstall process, Added Genome and wget for Linux installation and .bashrc file change. 

## Description
This Pull request has following features,

1.  As part Linux uninstallation process user can remove python binaries.
2.  Added Ansys-python-Manager launch command to .bashrc file.
3.  Added Gnome and wget as one of the dependencies in the installation process to support wsl. 
4.  Added pre-compiled python version NOTE/Warning on the installer

## Related Issue
https://github.com/ansys/python-installer-qt-gui/issues/238
https://github.com/ansys/python-installer-qt-gui/issues/241

### CODE REVIEW

File: doc/source/installer.rst [hunk 1]
```diff
@@ -71,12 +71,17 @@ Installing the ``Ansys Python Manager``
...
-             uninstallation, mark ``Delete virtual environment`` checkbox
+           * If you want to remove all virtual environments which was created by
+             Ansys Python Manageras part of uninstallation, mark
+             ``Delete virtual environment`` checkbox
 
-           * If you want to remove all the configurations as part of
+           * If you want to remove all configurations as part of
              uninstallation, mark ``Delete configurations`` checkbox
 
+           * If you want to remove all python installations which was installed by
```
Comment (RobPasMue) [inline doc/source/installer.rst:81]:
```suggestion
           * If you want to remove all Python installations which were installed by
```

File: doc/source/installer.rst [hunk 2]
```diff
@@ -71,12 +71,17 @@ Installing the ``Ansys Python Manager``
...
+             Ansys Python Manageras as part of uninstallation, mark
+             ``Delete python installations`` checkbox
```
Comment (RobPasMue) [inline doc/source/installer.rst:83]:
```suggestion
             ``Delete Python installations`` checkbox
```

File: doc/source/installer.rst [hunk 3]
```diff
@@ -130,12 +135,17 @@ Installing the ``Ansys Python Manager``
...
```
Comment (RobPasMue) [inline doc/source/installer.rst:147]:
```suggestion
             ``Delete Python installations`` checkbox
```

File: doc/source/installer.rst [hunk 4]
```diff
@@ -180,7 +190,7 @@ Installing the ``Ansys Python Manager``
              unzip Ansys-Python-Manager_linux_fedora_*.zip
              ./installer_Fedora.sh
 
-        #. Search for the ``Ansys Python Manager`` and run it.
+        #. Restart the machine and run ``ansys_python_installer``.
```
Comment (RobPasMue) [inline doc/source/installer.rst:193]:
```suggestion
        #. Restart the machine and run ``ansys_python_manager``.
```

File: src/ansys/tools/installer/constants.py [hunk 5]
```diff
@@ -68,6 +68,10 @@
 
 While choosing the latest version of Python is generally recommended, some third-party libraries and applications may not yet be fully compatible with the newest release. Therefore, it is recommended to try the second newest version, as it will still have most of the latest features and improvements while also having broader support among third-party packages."""
 
+PRE_COMPILED_PYTHON_WARNING = """
+
+NOTE - Only 'Python 3.11' version is readily available. Other python versions are compiled from source and it takes approx 2-3 minutes."""
```
Comment (RobPasMue) [inline src/ansys/tools/installer/constants.py:73]:
```suggestion
NOTE - Only 'Python 3.11' version is readily available. Other Python versions are compiled from source and it takes approximately 2-3 minutes."""
```

File: src/ansys/tools/installer/linux_functions.py [hunk 6]
```diff
@@ -139,10 +139,15 @@ def install_python_linux(filename):
...
 
 
-def find_miniforge_linux():
+def find_miniforge_linux(ansys_manager_installed_only=False):
     """
     Find miniforge installation on the host machine.
 
+    Parameters:
+    ----------
+    ansys_manager_installed_only : Boolean(default: False)
```
Comment (RobPasMue) [inline src/ansys/tools/installer/linux_functions.py:148]:
```suggestion
    ansys_manager_installed_only : bool, optional
```

File: src/ansys/tools/installer/uninstall.py [hunk 7]
```diff
@@ -86,6 +88,28 @@ def __init__(self, parent):
                 self.uninstall_window_cache_remove_venv_checkbox
             )
 
+            # remove installed python
+            uninstall_window_cache_remove_python_layout = QtWidgets.QHBoxLayout()
+            uninstall_window_cache_remove_python_text = QtWidgets.QLabel()
+            uninstall_window_cache_remove_python_text.setText(
+                "Delete python installations"
```
Comment (RobPasMue) [inline src/ansys/tools/installer/uninstall.py:95]:
```suggestion
                "Delete Python installations"
```

**Review by RobPasMue — CHANGES_REQUESTED** [review]

File: src/ansys/tools/installer/uninstall.py [hunk 8]
```diff
@@ -155,6 +183,9 @@ def _uninstall(self):
         if self.uninstall_window_cache_remove_venv_checkbox.isChecked():
```
[the remark this answers is not in the document]
Comment (anssakthi) [reply src/ansys/tools/installer/uninstall.py:183]: Nice catch @RobPasMue we forgot to include this in uninstaller script. Thanks

File: src/ansys/tools/installer/installed_table.py [hunk 9]
```diff
@@ -481,7 +481,7 @@ def delete_virtual_environment(self, point):
 
         # Create the context menu
         menu = QtWidgets.QMenu(self)
-        delete_action = menu.addAction("Delete virtual environment")
+        delete_action = menu.addAction("Delete virtual environments")
```
Comment (RobPasMue) [inline src/ansys/tools/installer/installed_table.py:484]:
```suggestion
        delete_action = menu.addAction("Delete virtual environment")
```

File: src/ansys/tools/installer/main.py [hunk 10]
```diff
@@ -229,14 +230,22 @@ def __init__(self, show=True):
...
         self.python_version_select.setCurrentIndex(default_index)
         python_version_layout.addWidget(self.python_version_select)
 
         python_version_box_layout.addWidget(python_version)
         form_layout.addWidget(python_version_box)
 
+        # Linux Warning/Note for Python pre-compiled version
+        if is_linux_os():
+            python_version_warning = QtWidgets.QLabel(PRE_COMPILED_PYTHON_WARNING)
+            python_version_warning.setStyleSheet("color: red;")
```
Comment (RobPasMue) [inline src/ansys/tools/installer/main.py:244]:
```suggestion
```

I would not add color to the note.

**Review by RobPasMue — APPROVED** [review]

**Comment by RobPasMue** [comment]
All good - this PR can be merged once CI/CD passes

**Pull request merged**